File: 30-days-of-code/java/day-16-exceptions.java
import java.io.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String S = in.next();
        // Begin Solution
        //---------------------------------------------------------------------------------------
        try {
            int converted = Integer.parseInt(S);
            System.out.println(converted);
        } catch(NumberFormatException e) {
            System.out.println("Bad String");
        }
        //---------------------------------------------------------------------------------------
        // End Solution
    }
}

File: 30-days-of-code/cpp/day-16-exceptions.cpp
#include <iostream>

using namespace std;


int main() {
    string S;
    cin >> S;
    // Begin Solution
    //---------------------------------------------------------------------------------------
    try {
        int converted = stoi(S);
        cout << S << endl;
    } catch(invalid_argument e) {
        cout << "Bad String" << endl;
    }
    //---------------------------------------------------------------------------------------
    // End Solution
    return 0;
}